flang/lib/Semantics/resolve-names.cpp

```diff
@@ -1605,6 +1609,26 @@ void OmpVisitor::Post(const parser::OpenMPBlockConstruct &x) {
   }
 }
 
+bool OmpVisitor::Pre(const parser::OpenMPDeclareMapperConstruct &x) {
+  AddOmpSourceRange(x.source);
+  BeginDeclTypeSpec();
+  PushScope(Scope::Kind::OtherConstruct, nullptr);
```

Comment:
I think the `PushScope` should be after the `MapperName` decl, as we want the scope of the Mapper to be the _curScope_, but the `var` declared within the mapper should have a scope of just this mapper directive.

Reply:
Done!